File: AKKeychainManager/iOS/DemoObjC/DemoObjC/ViewController.m
//
//  ViewController.m
//  DemoObjC
//
//  Created by Amr Koritem on 13/10/2022.
//

#import "ViewController.h"
@import AKKeychainManager;

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UILabel *label;
@property (weak, nonatomic) IBOutlet UITextField *textField;
@property (weak, nonatomic) IBOutlet UILabel *errorLabel;

@end

@implementation ViewController

- (IBAction)save {
    NSError *error = nil;
    NSString *savedData = [AKKeychainManager.shared loadWithService:@"test" account:@"test" error:&error];
    if (error == nil && savedData != nil) {
        [AKKeychainManager.shared updateWithService:@"test" account:@"test" value:_textField.text error:&error];
        [self handleError:error :^{}];
    } else {
        [self saveData];
    }
}

- (IBAction)show {
    NSError *error = nil;
    NSString *savedData = [AKKeychainManager.shared loadWithService:@"test" account:@"test" error:&error];
    [self handleError:error :^{
        self->_label.text = savedData;
    }];
}

- (void)showError:(NSString *)error {
    _errorLabel.text = error;
}

- (void)saveData {
    NSError *error = nil;
    [AKKeychainManager.shared saveWithService:@"test" account:@"test" value:_textField.text error:&error];
    [self handleError:error :^{}];
}

- (void)handleError:(NSError *)error :(void (^)(void))handler {
    if (error == nil) {
        handler();
        [self showError:@""];
    } else {
        [self showError:error.localizedDescription];
    }
}


@end

File: AKKeychainManager/tvOS/DemoSwift/DemoSwift/ViewController.swift
//
//  ViewController.swift
//  DemoSwift
//
//  Created by Amr Koritem on 27/02/2023.
//

import UIKit
import AKKeychainManager

class ViewController: UIViewController {
    @IBOutlet weak var label: UILabel!
    @IBOutlet weak var textField: UITextField!
    @IBOutlet weak var errorLabel: UILabel!
    
    @IBAction func save() {
        do {
            let _ = try AKKeychainManager.shared.load(service: "test", account: "test")
            try AKKeychainManager.shared.update(service: "test", account: "test", value: textField.text ?? "")
            showError("")
        } catch {
            guard let kError = error as? KeychainError else {
                showError(error.localizedDescription)
                return
            }
            switch kError {
            case .loadFailed:
                saveData()
            default:
                showError(kError.localizedDescription)
            }
        }
    }

    @IBAction func show() {
        do {
            let savedData = try AKKeychainManager.shared.load(service: "test", account: "test")
            label.text = savedData
            showError("")
        } catch {
            showError(error.localizedDescription)
        }
    }

    func showError(_ error: String) {
        errorLabel.text = error
    }

    func saveData() {
        do {
            try AKKeychainManager.shared.save(service: "test", account: "test", value: textField.text ?? "")
            showError("")
        } catch {
            showError(error.localizedDescription)
        }
    }
}

